libs/SmartSync/SmartSync/Classes/Util/SFSyncUpTarget.m

```diff
@@ -58,7 +58,7 @@ - (instancetype)init {
 #pragma mark - Serialization and factory methods
 
 + (instancetype)newFromDict:(NSDictionary*)dict {
-    // We should have an implementation class unless sync up was created with SDK before 5.1 and is not custom
+    // We should have an implementation class or a target type
```

Comment:
Comment was not quite accurate - hybrid apps might not specify implementation class